### ISSUE TITLE
Creating an OType.ANY property fails if a record already contains data for that property

### ISSUE THREAD

**Issue by peturrun** [orientechnologies/orientdb#7765]
## Expected behavior  
Since a property of type OType.ANY can contain objects of any type, it should be possible to create such a property regardless of what schema-less data is already present.

## Actual behavior  
Attempting to create the property fails:
```
Exception in thread "main" com.orientechnologies.orient.core.exception.OSchemaException: The database contains some schema-less data in the property 'MyType.myprop' that is not compatible with the type ANY. Fix those records and change the schema again
	DB name="test"
	at com.orientechnologies.orient.core.metadata.schema.OClassImpl.checkPersistentPropertyType(OClassImpl.java:2188)
	at com.orientechnologies.orient.core.metadata.schema.OClassImpl.addPropertyInternal(OClassImpl.java:1812)
	at com.orientechnologies.orient.core.metadata.schema.OClassImpl$6.call(OClassImpl.java:2606)
	at com.orientechnologies.orient.core.metadata.schema.OClassImpl$6.call(OClassImpl.java:2603)
	at com.orientechnologies.orient.core.db.OScenarioThreadLocal.executeAsDistributed(OScenarioThreadLocal.java:70)
	at com.orientechnologies.orient.core.metadata.schema.OClassImpl.addProperty(OClassImpl.java:2603)
	at com.orientechnologies.orient.core.metadata.schema.OClassImpl.createProperty(OClassImpl.java:807)
	at com.orientechnologies.orient.core.metadata.schema.OClassAbstractDelegate.createProperty(OClassAbstractDelegate.java:165)
	at com.tinkerpop.blueprints.impls.orient.OrientElementType.access$201(OrientElementType.java:35)
	at com.tinkerpop.blueprints.impls.orient.OrientElementType$3.call(OrientElementType.java:70)
	at com.tinkerpop.blueprints.impls.orient.OrientElementType$3.call(OrientElementType.java:67)
	at com.tinkerpop.blueprints.impls.orient.OrientBaseGraph.executeOutsideTx(OrientBaseGraph.java:1766)
	at com.tinkerpop.blueprints.impls.orient.OrientElementType.createProperty(OrientElementType.java:67)
	at com.tinkerpop.blueprints.impls.orient.OrientVertexType.createProperty(OrientVertexType.java:119)
	at com.activitystream.AnyTest.main(AnyTest.java:23)
```

## Steps to reproduce  
This program shows the problem:
```
import com.orientechnologies.orient.core.metadata.schema.OType;
import com.tinkerpop.blueprints.impls.orient.*;

public class AnyTest {

    public static void main(String[] args) {

        String url = "memory:test";

        OrientGraphNoTx noTx = new OrientGraphNoTx(url);
        OrientVertexType myType = noTx.createVertexType("MyType");

        // If the property is created first, strings can be inserted into it later...
        //myType.createProperty("myprop", OType.ANY);

        OrientGraph tx = new OrientGraph(url);
        tx.addVertex("class:MyType", "myprop", "myval");
        tx.commit();

        // ...but it is not possible to create the property if strings have been inserted first
        myType.createProperty("myprop", OType.ANY);
    }
}
```

**Comment by luigidellaquila** [comment]
Hi @peturrun 

Thank you for reporting.
@tglman I guess we can just skip these checks in case of Type.ANY, WDYT?
A check in `OClassImpl.checkPersistentPropertyType()` should be enough.

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @peturrun 

I pushed a fix for this problem on 2.2.x, the fix will be released with 2.2.28

Thanks

Luigi

**Comment by peturrun** [comment]
Thank you!
